### PR TITLE
[sailfishos][gecko] Remove nsWidget.h/cpp and mozqwidget.h/cpp from widget/qt. Contributes to JB#54190

### PR DESCRIPTION
**Pull request by chriadam**
Add the required CreateTopLevelWindow() and CreateChildWindow()
implementations to nsWindow.cpp

### CODE REVIEW

File: rpm/0002-sailfishos-qt-Bring-back-Qt-layer.-JB-50505.patch [hunk 1]
```diff
@@ -1,4 +1,4 @@
-From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From cd7fc15a1d7f3315c6c34635c6da3a2fd704c653 Mon Sep 17 00:00:00 2001
```
Comment (rainemak) [inline rpm/0002-sailfishos-qt-Bring-back-Qt-layer.-JB-50505.patch:1]: Format with --zero-commit

File: rpm/0002-sailfishos-qt-Bring-back-Qt-layer.-JB-50505.patch [hunk 2]
```diff
@@ -221,124 +221,122 @@ See upstream commit: a7c687ab305f41b3129191dc96682873d984c2ee
 
 Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
 Signed-off-by: Pavel Tumakaev <p.tumakaev@omprussia.ru>
+
+[sailfishos][qt] Remove nsWindow.h/cpp, mozqwidget.h/cpp. Contributes to JB#54190
```
Comment (rainemak) [inline rpm/0002-sailfishos-qt-Bring-back-Qt-layer.-JB-50505.patch:225]: Move this above Signed-offs

**Comment by rainemak** [comment]
Rebased

File: rpm/0002-sailfishos-qt-Bring-back-Qt-layer.-JB-50505.patch (hunk 1 above)
Comment (chriadam) [reply rpm/0002-sailfishos-qt-Bring-back-Qt-layer.-JB-50505.patch]: Fixed

File: rpm/0002-sailfishos-qt-Bring-back-Qt-layer.-JB-50505.patch (hunk 2 above)
Comment (chriadam) [reply rpm/0002-sailfishos-qt-Bring-back-Qt-layer.-JB-50505.patch]: Fixed

**Comment by chriadam** [comment]
Also removed nsQtKeyUtils.h/cpp as those are no longer required either.

**Comment by chriadam** [comment]
Hrm, I broke another patch due to the moz.build change.  Will fix.

**Comment by chriadam** [comment]
Fixed moz.build patch issue.

File: embedding/embedlite/embedshared/nsWindow.cpp [hunk 3]
```diff
@@ -387,3 +387,20 @@ nsWindow::CreateGLContextEarly(EmbedLiteWindowListener *aListener)
 
 }  // namespace embedlite
 }  // namespace mozilla
+
```
Comment (rainemak) [inline embedding/embedlite/embedshared/nsWindow.cpp:390]: Extra empty line.

Comment (chriadam) [reply embedding/embedlite/embedshared/nsWindow.cpp:390]: Fixed

**Review by rainemak — APPROVED** [review]
This is good now.

**Review by okodron — APPROVED** [review]

**Pull request merged**